tests/test_mozilla_reps.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Santiago Dueñas <sduenas@bitergia.com>
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import argparse
+import json
+import shutil
+import sys
+import tempfile
+import unittest
+
+import httpretty
+
+if not '..' in sys.path:
+    sys.path.insert(0, '..')
+
+from perceval.cache import Cache
+from perceval.errors import CacheError
+from perceval.backends.mozilla_reps import MozillaReps, MozillaRepsCommand, MozillaRepsClient
+
+
+MOZILLA_REPS_SERVER_URL = 'http://example.com'
+MOZILLA_REPS_API = MOZILLA_REPS_SERVER_URL + '/api/v1/'
+MOZILLA_REPS_API_EVENTS = MOZILLA_REPS_SERVER_URL + '/api/v1/event/'
+
+
+def read_file(filename, mode='r'):
+    with open(filename, mode) as f:
+        content = f.read()
+    return content
+
+
+class HTTPServer():
+
+    requests_http = []  # requests done to the server
+
+    @classmethod
+    def routes(cls, empty=False):
+        """Configure in http the routes to be served"""
+
+        mozilla_events = read_file('data/mozilla_reps_events.json')
+        if empty:
+            mozilla_events = read_file('data/mozilla_reps_events_empty.json')
+
+        def request_callback(method, uri, headers):
+            body = mozilla_events
+
+            HTTPServer.requests_http.append(httpretty.last_request())
+
+            return (200, headers, body)
+
+        httpretty.register_uri(httpretty.GET,
+                               MOZILLA_REPS_API_EVENTS,
+                               responses=[
+                                    httpretty.Response(body=request_callback)
+                               ])
+
+    @classmethod
+    def check_events_contents(cls, testObj, events):
```

Comment:
This should be outside this in a base class for testing or similar. It is not a good practice to pass the same class you're testing to get access to `unittest` methods.